slack-api.c

```diff
@@ -46,7 +46,7 @@ static void api_run(SlackAccount *sa);
 static void api_cb(PurpleUtilFetchUrlData *fetch, gpointer data, const gchar *buf, gsize len, const gchar *error) {
 	SlackAccount *sa = data;
 	SlackAPICall *call = g_queue_pop_head(&sa->api_calls);
-	g_return_if_fail(call->fetch == fetch || (call->fetch == NULL && error));
+	g_return_if_fail(call == NULL || (call->fetch == NULL && error) || call->fetch == fetch);
```

Comment:
I'm a little confused by this.  This is `g_return_if_fail` so it's basically now:
```
if (!(call == NULL || (call->fetch == NULL && error) || call->fetch == fetch))
   return;
call->fetch = NULL
```
So if `call == NULL`, we don't return, and instead immediatly try to derefence call.  Doesn't that then just crash as well?

Also what can it matter what order the other two terms happen in?  I feel like I'm missing something, or something else is going on here.

Reply:
To be clear, I think adding a `call != NULL && ...` here would make sense as a sanity check.

However, now that I'm looking, I think there something wrong with how the ratelimiting retry is done. I'll do a closer review there.

Reply:
I'm sorry, you're most likely right, this backwards thinking (negative check) had me confused, plus there is no easy way for me to reproduce the issue so the patches were never shown to be working.

The reordering was likely pointless, I did it out of habit, “checking for NULL pointers before using them”, but you're right, in this case the “usage” is just another comparison, so no segmentation fault would be caused by it.

Commit fb766f90f67e6a7280eb3f71104baff0e6721ed6 looks good to me, thanks for reworking the patch.  I'll soon apply it to my setup and report any new crash.